### ISSUE TITLE
Can't upload files larger than 2.1 GB

### ISSUE THREAD

**Issue by dansmith01** [cloudyr/aws.s3#274]
I can't upload a file larger than about 2.1 GB using `put_object`. I suspect the issue is line 287 of [put_object.R](https://github.com/cloudyr/aws.s3/blob/master/R/put_object.R), where the file size is converted to integer. Any file size larger than the max integer range (`.Machine$integer.max` or 2147483647) results in `calculate_data_size()` returning `NA`.

I noticed that that conversion to integer is a [recent modification from Sept 3](https://github.com/cloudyr/aws.s3/commit/86b16545d57c509452e26a9d070c295510a0788c) meant to address an issue with canonical headers being signed properly.

```R
## load package
library("aws.s3")


## Create a 2.2 Gb file
saveRDS(rep(get("volcano"), 55000), "myfile.rds", compress=FALSE)


## Try to upload it
aws.s3::put_object("myfile.rds", "uploaded.rds", "somebucket")
Error in if (headers[["Content-Length"]] > 1e+07) { :
  missing value where TRUE/FALSE needed
In addition: Warning message:
In calculate_data_size(file) : NAs introduced by coercion to integer range


## Try to upload it with the multipart flag
aws.s3::put_object("myfile.rds", "uploaded.rds", "somebucket", multipart=TRUE)
Error in if (size < partsize) { : missing value where TRUE/FALSE needed
In addition: Warning message:
In calculate_data_size(file) : NAs introduced by coercion to integer range


## Session info
sessionInfo()
R version 3.3.1 (2016-06-21)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Red Hat Enterprise Linux Server release 6.5 (Santiago)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C
 [9] LC_ADDRESS=C               LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
[1] aws.s3_0.3.20

loaded via a namespace (and not attached):
[1] httr_1.3.1          R6_2.3.0            tools_3.3.1
[4] base64enc_0.1-3     curl_3.2            Rcpp_0.12.19
[7] aws.signature_0.4.4 xml2_1.2.0.9000     digest_0.6.18
```

**Comment by dansmith01** [comment]
Just noticed this is a duplicate of issue #264 and will be resolved by pull request #273 .

**Comment by acolum** [comment]
This issue has been fixed with PR #273.
